File: src/main/Scala/Task2_A_1.py
from pyspark.sql import SparkSession

spark = SparkSession \
    .builder \
    .appName("Python Spark SQL basic example") \
    .config("spark.some.config.option", "some-value") \
    .getOrCreate()

sc = spark.sparkContext

customers_schema = """
    ID INT,
    Name STRING,
    Age INT,
    CountryCode INT,
    Salary FLOAT
"""

purchases_schema = """
    TransID INT,
    CustID INT,
    TransTotal FLOAT,
    TransNumItems INT,
    TransDesc STRING
"""

purchpath = "data/purchases.txt"
custpath = "data/customers.txt"
purchdf = spark.read.csv(purchpath, header=False, schema=purchases_schema)
custdf = spark.read.csv(custpath, header=False, schema=customers_schema)

T1_df = purchdf.filter(purchdf['TransTotal'] > 600)

T1_df.createOrReplaceTempView("T1")

T1_df.show(n=100)

spark.stop()

File: src/main/Scala/Task2_A_1.scala
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.types._

object Task2_A_1 {

  def main(args: Array[String]): Unit = {
    val spark = SparkSession.builder().appName("Task2_A_2").master("local").getOrCreate()
    val customerSchema = new StructType()
      .add("id", IntegerType)
      .add("name", StringType)
      .add("age", IntegerType)
      .add("countryCode", IntegerType)
      .add("salary", FloatType)

    val purchaseSchema = new StructType()
      .add("transID", IntegerType)
      .add("custID", IntegerType)
      .add("transTotal", FloatType)
      .add("transNumItems", IntegerType)
      .add("transDesc", StringType)

    val customersDF = spark.read
      .schema(customerSchema)
      .csv("data/customers.txt")
    val purchasesDF = spark.read
      .schema(purchaseSchema)
      .csv("data/purchases.txt")

    customersDF.createOrReplaceTempView("customers")
    purchasesDF.createOrReplaceTempView("purchases")

    val t1 = spark.sql("SELECT * FROM purchasesT WHERE TransTotal > 600")

    t1.coalesce(1).write.option("header",true).csv("data/T1.csv")

    t1.show()

    spark.stop()
  }
}